dart/lib/src/sentry_client.dart

```diff
@@ -462,18 +454,43 @@ class SentryClient {
     _options.recorder.recordLostEvent(reason, category);
   }
 
-  SentryEvent _eventWithRemovedBreadcrumbsIfHandled(SentryEvent event) {
+  SentryEvent _eventWithoutBreadcrumbsIfNeeded(SentryEvent event) {
+    if (_shouldRemoveBreadcrumbs(event)) {
+      return event.copyWith(breadcrumbs: []);
+    } else {
+      return event;
+    }
+  }
+
+  SentryTransaction _transactionWithoutBreadcrumbsIfNeeded(
+      SentryTransaction transaction) {
+    if (_shouldRemoveBreadcrumbs(transaction)) {
+      return transaction.copyWith(breadcrumbs: []);
+    } else {
+      return transaction;
+    }
+  }
```

Comment:
Would be possible to use the base class here? `SentryTransaction` extends `SentryEvent` and `breadcrumbs` is part of `SentryEvent`